reviews/models.py

```diff
@@ -143,6 +143,14 @@ def get_route_display(self):
 
     def accountable_user(self):
         return self.proposal.accountable_user()
+    
+    def get_applicant_names_emails(self):
+        names = [user.get_full_name() for user in self.proposal.applicants.all()]
+        emails = [user.email for user in self.proposal.applicants.all()]
+        name_email_string = []
+        for i in range(len(names)):
+            name_email_string.append(",\n".join([names[i], emails[i]]))
```

Comment:
I'm not a fan of hitting the database twice to ask for `proposal.applicants.all()`. Also, this leaves room for bugs if the ordering or length of either request is different, however unlikely. I would suggest:

```python3
names_and_emails = [
    (user.get_full_name(), user.email)
    for user in proposal.applicants.all()
]
name_email_string = []
for pair in names_and_emails:
    name_email_string.append(
        ",\n".join(pair)
    )
return name_email_string
```

Like that you only hit the database once, and the name and email are guaranteed to be from the same user.

I did enjoy that `name_email_string` is, in fact, a list, so I left that in.

Reply:
Ah you are right of coure, I implemented your solution. I also updated the name to `name_email_strings`, just to make it a tiny bit easier to understand.